ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/LedgerTestContext.scala

```diff
@@ -43,9 +44,14 @@ private[testtool] final class LedgerTestContext private[infrastructure] (
       .sequence(allocation.partyCounts.map(partyCount => {
         val participant = nextParticipant()
         for {
-          parties <- participant.allocateParties(partyCount.count)
-          partiesSet = parties.toSet
-          _ <- participant.waitForParties(participants, partiesSet)
+          parties <- if (openWorld) {
```

Comment:
It's probably worth moving this into a method rather than nesting this much.

Reply:
Good call. I also realized, in doing this, that it doesn't make sense to have both `openWorld` and `waitForParties`, so I moved them into an enumeration (sealed trait and case objects), `PartyAllocationConfiguration`, that makes it impossible to have both.